packages/pug/lib/index.js

```diff
@@ -50,6 +50,15 @@ function applyPlugins(value, options, plugins, name) {
   }, value);
 }
 
+function getPlugin(plugins, name, defaultFunction) {
+  for (var plugin of plugins) {
+    if (plugin[name]) {
+      return plugin[name];
+    }
+  }
+  return defaultFunction;
```

Comment:
This function needs to throw an error if there is more than one plugin implementing the same `name`. i.e. this approach does not support multiple code-gen plugins.

Reply:
I decided to remove the new function and reuse `findReplacementFunc()` since it was almost doing what I needed.